mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/enricher/ExtensionsErrorsDeclarationEnricher.java

```diff
@@ -76,13 +79,30 @@ public WalkDelegateDelegateDeclaration(ExtensionLoadingContext extensionLoadingC
     public void onOperation(WithOperationsDeclaration owner, OperationDeclaration operationDeclaration) {
       errorModels.addAll(operationDeclaration.getErrorModels());
       if (operationDeclaration.isRequiresConnection()) {
-        addErrorModel(operationDeclaration, connectivityError, CONNECTIVITY_ERROR_TYPE);
-        addErrorModel(operationDeclaration, retryErrorModel, RETRY_EXHAUSTED_ERROR_TYPE);
+        if (requiresConnectivityError(extensionDeclaration)) {
+          addErrorModel(operationDeclaration, connectivityError, CONNECTIVITY_ERROR_TYPE);
+        }
+        if (requiresRetryExhaustedError(operationDeclaration)) {
+          addErrorModel(operationDeclaration, retryErrorModel, RETRY_EXHAUSTED_ERROR_TYPE);
+        }
       }
 
       assureValidationError(extensionDeclaration, operationDeclaration);
     }
 
+    private boolean requiresConnectivityError(ExtensionDeclaration extensionDeclaration) {
```

Comment:
replace this method by a more generic `isMuleAppExtensionModel()`. This method should be defined on some non-api utils

Reply:
unresolved